app/javascript/components/navbar/AskQuestion.tsx

```diff
@@ -2,82 +2,128 @@ import React, { useState, useContext } from 'react';
-  const url = Routes.ask_question_exam_path(examID);
-  const res = await fetch(url, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-      'X-CSRF-Token': getCSRFToken(),
-    },
-    credentials: 'same-origin',
-    body: JSON.stringify({
-      message: {
-        body: question,
-      },
-    }),
-  });
-  const json = await (res.json() as Promise<{success: boolean}>);
-  return json.success;
+interface ShowStatusProps {
+  status: ProfQuestionStatus;
 }
 
-const AskQuestion: React.FC<{}> = () => {
-  const [val, setVal] = useState('');
-  const [error, setError] = useState('');
-  const [variant, setVariant] = useState<AlertProps['variant']>('danger');
-  const [showAlert, setShowAlert] = useState(false);
-  const [saving, setSaving] = useState(false);
-  const buttonText = saving ? 'Saving...' : 'Submit';
-  const { railsExam } = useContext(RailsContext);
+const ShowStatus: React.FC<ShowStatusProps> = (props) => {
+  const {
+    status,
+  } = props;
+  const size = '1.5em';
+  switch (status) {
+    case 'SENDING':
+      return (
+        <span className="text-info">
+          <Spinner
+            size="sm"
+            animation="border"
+          />
+        </span>
+      );
+    case 'FAILED':
+      return (
+        <span className="text-danger">
+          <AiFillCloseCircle
+            size={size}
+          />
+        </span>
+      );
+    case 'SENT':
+      return (
+        <span className="text-success">
+          <AiFillCheckCircle
+            size={size}
+          />
+        </span>
+      );
+    default:
+      throw new ExhaustiveSwitchError(status);
+  }
+};
+
+interface ShowQuestionProps {
+  question: ProfQuestion;
+}
+
+const ShowQuestion: React.FC<ShowQuestionProps> = (props) => {
+  const {
+    question,
+  } = props;
   return (
     <>
+      <div>
+        <div className="mr-1 d-inline">
+          <ShowStatus status={question.status} />
+        </div>
+        <i>{question.time.toLocaleString(DateTime.TIME_SIMPLE)}</i>
+      </div>
+      <p>{question.body}</p>
+    </>
+  );
+};
+
+interface AskQuestionProps {
+  questions: ProfQuestion[];
+  onSubmit: (examID: number, body: string) => void;
+}
+
+const AskQuestion: React.FC<AskQuestionProps> = (props) => {
+  const {
+    questions,
+    onSubmit,
+  } = props;
+  const {
+    railsExam,
+  } = useContext(RailsContext);
+
+  const anySending = questions.some((q) => q.status === 'SENDING');
+
+  const [val, setVal] = useState('');
+  const buttonText = anySending ? 'Saving...' : 'Submit';
+  return (
+    <div>
       <Form.Control
         value={val}
         onChange={(event): void => {
           setVal(event.target.value);
         }}
         as="textarea"
-        disabled={saving}
+        disabled={anySending}
       />
       <Button
         className="ml-auto mt-3"
         variant="success"
-        disabled={saving}
+        disabled={anySending}
```

Comment:
This should be disabled until the question has non-empty text, too.  Currently, I can click Submit on an empty question, and get a X response for it, which is a bit weird...